core/src/test/scala/integration/kafka/admin/ConfigCommandIntegrationTest.scala

```diff
@@ -162,19 +174,198 @@ class ConfigCommandIntegrationTest extends QuorumTestHarness with Logging {
+      "Timeout waiting for topic config propagating to broker")
+
+    val describeResult = TestUtils.grabConsoleOutput(
+      ConfigCommand.describeConfig(cluster.createAdminClient(), new ConfigCommandOptions(
+        Array("--bootstrap-server", s"${cluster.bootstrapServers()}",
+          "--describe",
+          "--entity-type", "topics",
+          "--entity-name", "test-config-topic")
+      )))
+    assertTrue(describeResult.contains("segment.bytes=10240000"))
+    assertTrue(describeResult.contains("sensitive=false"))
+
+    ConfigCommand.alterConfig(cluster.createAdminClient(), new ConfigCommandOptions(
+      Array("--bootstrap-server", s"${cluster.bootstrapServers()}",
+        "--alter",
+        "--delete-config", "segment.bytes",
+        "--entity-type", "topics",
+        "--entity-name", "test-config-topic")
+    ), true)
+    TestUtils.waitUntilTrue(
+      () => cluster.brokers().collect(Collectors.toList[KafkaBroker]).asScala.forall(broker => broker.logManager.logsByTopic("test-config-topic").head.config.getInt("segment.bytes") != 10240000),
+      "Timeout waiting for topic config propagating to broker")
+
+    assertFalse(TestUtils.grabConsoleOutput(
+      ConfigCommand.describeConfig(cluster.createAdminClient(), new ConfigCommandOptions(
+        Array("--bootstrap-server", s"${cluster.bootstrapServers()}",
+          "--describe",
+          "--entity-type", "topics",
+          "--entity-name", "test-config-topic")
+      ))).contains("segment.bytes"))
+  }
+
+  @ClusterTest
+  def testUpdateBrokerConfigNotAffectedByInvalidConfig(): Unit = {
+    // Test case from KAFKA-13788
+    ConfigCommand.alterConfig(cluster.createAdminClient(), new ConfigCommandOptions(
+      Array("--bootstrap-server", s"${cluster.bootstrapServers()}",
+        "--alter",
+        "--add-config", "log.cleaner.threadzz=2",
+        "--entity-type", "brokers",
+        "--entity-default")
+    ), true)
+
+    ConfigCommand.alterConfig(cluster.createAdminClient(), new ConfigCommandOptions(
+      Array("--bootstrap-server", s"${cluster.bootstrapServers()}",
+        "--alter",
+        "--add-config", "log.cleaner.threads=2",
+        "--entity-type", "brokers",
+        "--entity-default")
+    ), true)
+  }
+
+  // TODO this test doesn't make sense because we can't produce `UnsupportedVersionException` by setting inter.broker.protocol.version
+  @ClusterTest(clusterType=Type.ZK, metadataVersion = MetadataVersion.IBP_3_2_IV0)
+  def testFallbackToDeprecatedAlterConfigs(): Unit = {
+    ConfigCommand.alterConfig(cluster.createAdminClient(), new ConfigCommandOptions(
+      Array("--bootstrap-server", s"${cluster.bootstrapServers()}",
+        "--alter",
+        "--add-config", "log.cleaner.threads=2",
+        "--entity-type", "brokers",
+        "--entity-default")
+    ), true)
+  }
```

Comment:
I can't see any tests for `useIncrementalAlterConfigs` set to false. Can we test this path as well if possible?

Reply:
This case doesn't make sense because we can't produce `UnsupportedVersionException` by setting inter.broker.protocol.version, so I'm waiting for help from other reviewers, and just leave it unchanged for now.